dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -133,4 +284,53 @@ public void test_Publish_With_Save_Contentlet_Actionlet_Tag () throws DotSecurit
         Assert.assertTrue(contentletPublished.isLive());
     }
 
+    private void checkContentSaved(Contentlet contentletSaved) throws DotDataException {
+        Assert.assertNotNull(contentletSaved);
+        Assert.assertEquals("Test", contentletSaved.getStringProperty("title"));
+        Assert.assertEquals("Test", contentletSaved.getStringProperty("txt"));
+
+        final List<TagInode> tagInodes = APILocator.getTagAPI().getTagInodesByInode(contentletSaved.getInode());
+        Assert.assertNotNull(tagInodes);
+        Assert.assertFalse(tagInodes.isEmpty());
+
+        contentletSaved.setTags();
+    }
+
+    @NotNull
+    private static void addPermissionToAddChildren(final Role role, final ContentType contentType) throws DotDataException {
+
+        final Permission contentTypePermission = getPermission(role, contentType, PermissionLevel.WRITE.getType());
+
+        try {
+
+            APILocator.getPermissionAPI().save(contentTypePermission, contentType, systemUser, false);
+
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private static void addPermissionToSaveAction(final Role role) throws DotDataException {
+
+        final WorkflowAction action = FactoryLocator.getWorkFlowFactory().findAction(SystemWorkflowConstants.WORKFLOW_SAVE_ACTION_ID);
+        final Permission actionPermission = getPermission(role, action, PermissionLevel.USE.getType());
+
+        try {
+             APILocator.getPermissionAPI().save(actionPermission, action, systemUser, false);
+
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private static Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)